enzyme/Enzyme/EnzymeLogic.cpp

```diff
@@ -3321,12 +3501,29 @@ Function *EnzymeLogic::CreatePrimalAndGradient(
     }
 
     if (!key.returnUsed && key.freeMemory) {
+      auto res =
```

Comment:
Please use a structured binding to unwrap the pair elements: 
`auto&& [nameForFirstPairElement, nameForSecondPairElement] = ...`